modules/localstack/types.go

```diff
@@ -0,0 +1,265 @@
+package localstack
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/imdario/mergo"
+	"github.com/testcontainers/testcontainers-go"
+)
+
+// LocalStackContainer represents the LocalStack container type used in the module
+type LocalStackContainer struct {
+	testcontainers.Container
+	EnabledServices map[string]Service
+}
+
+// ServicePort returns the port of the given service
+func (l *LocalStackContainer) ServicePort(ctx context.Context, service EnabledService) (nat.Port, error) {
+	if _, ok := l.EnabledServices[service.Name()]; !ok {
+		return "", fmt.Errorf("service %s is not enabled", service.Name())
+	}
+
+	internalPort := l.EnabledServices[service.Name()].servicePort()
+
+	p, err := nat.NewPort("tcp", fmt.Sprintf("%d", internalPort))
+	if err != nil {
+		return "", err
+	}
+
+	return l.MappedPort(ctx, p)
+}
+
+// LocalStackContainerRequest represents the LocalStack container request type used in the module
+// to configure the container
+type LocalStackContainerRequest struct {
+	testcontainers.ContainerRequest
+	enabledServices []Service
+}
+
+// EnabledService is an interface for services that can be enabled
+type EnabledService interface {
+	Name() string
+	Named(string) string
+	Port() int
+}
+
+// Service represents a LocalStack service, such as S3, SQS, etc.
+type Service struct {
+	name string
+	port int
+}
+
+// Name returns the name of the service
+func (s Service) Name() string {
+	return s.name
+}
+
+// Named returns the name of the service with the given name
+func (s Service) Named(name string) string {
+	return name
+}
+
+// Port returns the port of the service
+func (s Service) Port() int {
+	return s.port
+}
+
+// servicePort returns the default port
+func (s Service) servicePort() int {
+	return defaultPort
+}
+
+// List of AWS services, ordered alphabetically
+
+// APIGateway is the API Gateway service
+var APIGateway = Service{
+	name: "apigateway",
+	port: 4567,
```

Comment:
if legacy mode is not supported then port can be remove it. If want to use it can override the ContainerRequest adding this port manually.

Reply:
We removed everything regarding Services